kpatch-build/create-diff-object.c

```diff
@@ -2156,7 +2156,7 @@ static bool should_keep_jump_label(struct lookup_table *lookup,
 		}
 	}
 
-	if (i != 3)
+	if (i != 3 || !key || !code)
```

Comment:
It's redundant to check for `key` and `code` value, when checking for `i != 3`. Which implicitly covers the NULL check of both variables.

Reply:
Right, we can deduce that from the loop code, but apparently the compiler can't for some reason.  /shrugs